event_listener/dfusion_db/event_pusher.py

```diff
@@ -69,11 +78,34 @@ def update_accounts(event: Dict[str, Union[int, str, str, int]]):
         db.accounts.insert_one(new_account_record)
 
     elif transition_type == 1:  # Withdraw
-        pass
+
+        requested_withdraws = db.withdraws.find({'slot': event['slot']})
+
+        for deposit in requested_withdraws:
+            a_id = deposit['accountId']
+            t_id = deposit['tokenId']
+            amount = deposit['amount']
+
+            index = num_tokens * (a_id - 1) + (t_id - 1)
+            # Balances are stored as [b(a1, t1), b(a1, t2), ... b(a1, T), b(a2, t1), ...]
+            if balances[index] - amount >= 0:
```

Comment:
I would prefer to check here also the bitmap from the snark, which will indicate whether a withdrawal was processed.

Reply:
Right although we are not emitting the bitmap in the transition event. This bitmap is constructed based on the account states at the time of apply withdrawal....

Reply:
I would argue that this is the correct logic since this is exactly how your driver code will construct the bitmap.